go-controller/pkg/ovn/policy.go

```diff
@@ -1377,53 +1326,49 @@ func (oc *Controller) handlePeerNamespaceSelectorOnUpdate(np *networkPolicy, gp
 
 func (oc *Controller) handlePeerNamespaceSelector(
 	namespaceSelector *metav1.LabelSelector,
-	gress *gressPolicy, np *networkPolicy) {
+	gress *gressPolicy, policyHandlerID uint64,
+	policy *knet.NetworkPolicy, portGroupName string) {
 
 	// NetworkPolicy is validated by the apiserver; this can't fail.
 	sel, _ := metav1.LabelSelectorAsSelector(namespaceSelector)
 
-	h := oc.watchFactory.AddFilteredNamespaceHandler("", sel,
+	oc.watchFactory.AddFilteredNamespaceHandler("", policyHandlerID, sel,
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
 				namespace := obj.(*kapi.Namespace)
 				// Update the ACL ...
-				oc.handlePeerNamespaceSelectorOnUpdate(np, gress, func() bool {
+				oc.handlePeerNamespaceSelectorOnUpdate(policy, portGroupName, gress, func() bool {
 					// ... on condition that the added address set was not already in the 'gress policy
 					return gress.addNamespaceAddressSet(namespace.Name)
 				})
 			},
 			DeleteFunc: func(obj interface{}) {
 				namespace := obj.(*kapi.Namespace)
 				// Update the ACL ...
-				oc.handlePeerNamespaceSelectorOnUpdate(np, gress, func() bool {
+				oc.handlePeerNamespaceSelectorOnUpdate(policy, portGroupName, gress, func() bool {
 					// ... on condition that the removed address set was in the 'gress policy
 					return gress.delNamespaceAddressSet(namespace.Name)
 				})
 			},
 			UpdateFunc: func(oldObj, newObj interface{}) {
 			},
 		}, func(i []interface{}) {
-			// This needs to be a write lock because there's no locking around 'gress policies
```

Comment:
@jcaamano you added this. I didn't really understand what you meant here, but I didn't see a reason we need to lock the network policy here. Can you take a look at this?

Reply:
Because it makes changes on the gress policy and gress policies are protected with the np lock they belong to.